spring-kafka/src/main/java/org/hitachivantara/spring/kafka/listener/AbstractMessageListener.java

```diff
@@ -0,0 +1,205 @@
+   * use a {@link org.springframework.kafka.listener.adapter.RetryingAcknowledgingMessageListenerAdapter}.
+   * @return
+   */
+  protected AcknowledgingMessageListener<K, V> getMessageListener() {
+    return this;
+  }
+
+  /**
+   * Assembles the listener. The container listener is also started here.
+   */
+  public void assembleListenerContainer() {
+    ContainerProperties containerProperties = buildContainerProperties();
+    this.messageListenerContainer = buildMessageListenerContainer( consumerFactory, containerProperties );
+  }
+
+  /**
+   * Builds the message listener to use in the assembleListenerContainer method.
+   * @param consumerFactory the consumer factory.
+   * @param containerProperties The container properties
+   * @return A {@link MessageListenerContainer} instance.
+   */
+  protected MessageListenerContainer buildMessageListenerContainer( ConsumerFactory<K, V> consumerFactory, ContainerProperties containerProperties ) {
+    return new KafkaMessageListenerContainer<>( consumerFactory, containerProperties );
+  }
+
+  /**
+   * Builds the container properties. This implementation uses the local topic variable.
+   * This container properties are used in the assembleListenerContainer method that is then passed to
+   * the buildMessageListenerContainer method.
+   *
+   * Both the local topic, callbackOnPartitionsRevoked and callbackOnPartitionsAssigned variables
+   * are used on this default implementation.
+   *
+   * The topic name is also used to set the listener group with the same value.
+   *
+   * The Ack mode is set to MANUAL_IMMEDIATE.
+   * @return A {@link ContainerProperties} instance.
+   */
+  protected ContainerProperties buildContainerProperties() {
+    ContainerProperties containerProperties = new ContainerProperties( topic );
+    containerProperties.setMessageListener( this.getMessageListener() );
+    containerProperties.setGroupId( topic );
+    containerProperties.setAckMode( AbstractMessageListenerContainer.AckMode.MANUAL_IMMEDIATE );
+    containerProperties.setConsumerRebalanceListener( new ConsumerRebalanceListener() {
+      @Override
+      public void onPartitionsRevoked( Collection<TopicPartition> collection ) {
+        if ( callbackOnPartitionsRevoked != null ) {
+          callbackOnPartitionsRevoked.run();
+        } else {
+          logger.debug( "Partition revoked for {}", topic );
+        }
+      }
+
+      @Override
+      public void onPartitionsAssigned( Collection<TopicPartition> collection ) {
+        if ( callbackOnPartitionsAssigned != null ) {
+          callbackOnPartitionsAssigned.run();
+        } else {
+          logger.debug( "Partition assigned for {}", topic );
+        }
+      }
+    } );
+    return containerProperties;
+  }
+
+
+  @Override
+  public void registerSeekCallback( ConsumerSeekCallback consumerSeekCallback ) {
+    logger.info( "Register seek callback was invoked... nothing will be done" );
```

Comment:
Any relevant information in `consumerSeekCallback` to add to the log message?

Reply:
We could take some information from the object received but I'm not sure what useful information it would have for an implementation that doesn't override this.